packages/fx-core/src/plugins/resource/simpleauth/plugin.ts

```diff
@@ -135,7 +135,55 @@ export class SimpleAuthPluginImpl {
 
     const configs = Utils.getWebAppConfig(ctx, false);
 
-    await this.webAppClient.configWebApp(configs);
+    if (isArmSupportEnabled()) {
+      const credentials = await ctx.azureAccountProvider!.getAccountCredentialAsync();
+      const subscriptionInfo = await ctx.azureAccountProvider?.getSelectedSubscription();
+      const subscriptionId = subscriptionInfo!.subscriptionId;
+      const resourceNameSuffix = Utils.getConfigValueWithValidation(
+        ctx,
+        Constants.SolutionPlugin.id,
+        Constants.SolutionPlugin.configKeys.resourceNameSuffix
+      ) as string;
+      const resourceGroupName = Utils.getConfigValueWithValidation(
+        ctx,
+        Constants.SolutionPlugin.id,
+        Constants.SolutionPlugin.configKeys.resourceGroupName
+      ) as string;
+      const location = Utils.getConfigValueWithValidation(
+        ctx,
+        Constants.SolutionPlugin.id,
+        Constants.SolutionPlugin.configKeys.location
+      ) as string;
+
+      const webAppName = Utils.generateResourceName(
+        ctx.projectSettings!.appName,
+        resourceNameSuffix
+      );
+      const appServicePlanName = webAppName;
+      this.webAppClient = new WebAppClient(
```

Comment:
Seems duplicated as above. Can we extract them in a method?

Reply:
it is fixed. thanks